### PR TITLE
feat/add_scheduelar_and_mailing_feature

### PR DESCRIPTION
**Pull request by Zenjjim**
# PR NAME
## Description
Link to Trello task: ...
https://trello.com/c/pyZd4LLm/206-send-schedule-mail

Comments/issues:
Create schedular with celery and mailer with django.core.mail. Also added mail and notification for signup, waitinglist, deadline, and evaluation

## Pull request checklist

Please check if your PR fulfills the following requirements:
- [x] PR name is descriptive
- [x] The PR includes the link to the relevant Trello task
- [ ] Tests for the changes have been added (for bug fixes / features)
- [ ] Docs have been reviewed and added / updated if needed (for bug fixes / features)

- [x] `python manage.py runserver` was run locally witout errors

## Pull request type

Please check the type of change your PR introduces:
- [ ] Bugfix
- [x] Feature
- [ ] Code style update (formatting, renaming)
- [ ] Refactoring (no functional changes, no api changes)
- [ ] Build related changes
- [ ] Documentation content changes
- [ ] Other (please describe):

### CODE REVIEW

File: app/content/templates/waitlist.html [hunk 1]
```diff
@@ -0,0 +1,29 @@
...
+            <o:OfficeDocumentSettings>
+            <o:AllowPNG/>
+            <o:PixelsPerInch>96</o:PixelsPerInch>
+            </o:OfficeDocumentSettings>
+        </xml>
+        <![endif]-->
+        <meta charset="UTF-8">
+        <meta http-equiv="X-UA-Compatible" content="IE=edge">
+        <meta name="viewport" content="width=device-width, initial-scale=1">
+        <title>TIHLDE - Du er påmeldt</title>
```
Comment (olros) [inline app/content/templates/waitlist.html]: Denne glemte jeg å endre til passende tittel. Det samme gjelder de andre templatene

File: app/celery.py [hunk 2]
```diff
@@ -0,0 +1,32 @@
...
+    accept_content=['json'],  # Ignore other content
+    result_serializer='json',
+    timezone='Europe/Oslo',
+    enable_utc=True,
+)
+
+
+@app.task(bind=True)
+def debug_task(self):
+    print('Request: {0!r}'.format(self.request))
```
Comment (Eirsteir) [inline app/celery.py]: Splendid! I assume this kind of feature will be hard to debug :)

File: app/content/models/event.py [hunk 3]
```diff
@@ -35,6 +45,11 @@ class Event(BaseModel, OptionalImage):
     sign_off_deadline = models.DateTimeField(blank=True, null=True, default=None)
 
     registration_priorities = models.ManyToManyField(Priority, blank=True, default=None, related_name='priorities')
+    evaluate_link = models.CharField(max_length=200, blank=True, null=True)
```
Comment (Eirsteir) [inline app/content/models/event.py]: What is evaluating what here? I am just curious ;)

File: app/content/models/event.py [hunk 4]
```diff
@@ -123,4 +138,13 @@ def check_start_time_is_after_end_time(self):
...
     def save(self, *args, **kwargs):
+        try:
+            revoke(self.end_date_schedular_id, terminate=True)
+            revoke(self.sign_off_deadline_schedular_id, terminate=True)
+            if self.evaluate_link and self.end_date < today() and not self.closed:
+                self.end_date_schedular_id = event_end_schedular.apply_async(eta=(self.end_date + timedelta(days=1)), kwargs={'pk': self.pk, 'title': self.title, 'date': self.start_date, 'evaluate_link':self.evaluate_link})
+            if self.sign_up and self.sign_off_deadline < today() and not self.closed:
+                self.sign_off_deadline_schedular_id = event_sign_off_deadline_schedular.apply_async(eta=(self.sign_off_deadline - timedelta(days=1)), kwargs={'pk': self.pk, 'title':self.title})
+        except Exception as e:
+            print(e)
```
Comment (Eirsteir) [inline app/content/models/event.py]: Remove this print. We should add logging instead

File: app/content/models/user_event.py [hunk 5]
```diff
@@ -33,6 +37,7 @@ def save(self, *args, **kwargs):
         if not self.user_event_id:
             return self.create(*args, **kwargs)
 
+        self.send_notification_and_mail()
```
Comment (Eirsteir) [inline app/content/models/user_event.py]: This will run twice when the user_event is being created

File: app/content/models/user_event.py [hunk 6]
```diff
@@ -77,17 +105,17 @@ def clean(self):
...
         self.check_registration_has_ended()
 
     def check_registration_has_started(self):
-        if self.event.start_registration_at > today():
+        if self.event.start_registration_at > today() and not self.user_event_id:
             raise ValidationError(_('The registration for this event has not started yet.'))
 
     def check_registration_has_ended(self):
-        if self.event.end_registration_at < today():
+        if self.event.end_registration_at < today() and not self.user_event_id:
```
Comment (Eirsteir) [inline app/content/models/user_event.py]: These are a bit redundant once you check for id when calling `validate_start_and_end_registration_time`

File: app/content/views/event.py [hunk 7]
```diff
@@ -31,6 +33,20 @@ def get_queryset(self):
...
+            event = Event.objects.get(pk=pk)
+            if is_admin_user(request):
+                serializer = EventAdminSerializer(event, context={'request': request}, many=False)
+            else:
+                serializer = EventSerializer(event, context={'request': request}, many=False)
+            return Response(serializer.data)
+        except Event.DoesNotExist:
+            return Response({'detail': _('User event not found.')}, status=404)
+        except Exception as e:
+            print(e)
```
Comment (Eirsteir) [inline app/content/views/event.py]: Remove print statements.

File: app/content/models/event.py (hunk 3 above)
Comment (Zenjjim) [reply app/content/models/event.py]: A link to a google form that will be sent out as mail after the event is done :)

File: app/content/models/user_event.py (hunk 5 above)
Comment (Zenjjim) [reply app/content/models/user_event.py]: It wont because self.create returns so it has no reason to go futher down

**Pull request merged**